modules/shapefile/src/shapefile-loader.js

```diff
@@ -0,0 +1,59 @@
+import {SHPLoader} from './shp-loader';
+/** @typedef {import('@loaders.gl/loader-utils').LoaderObject} LoaderObject */
+
+// __VERSION__ is injected by babel-plugin-version-inline
+// @ts-ignore TS2304: Cannot find name '__VERSION__'.
+const VERSION = typeof __VERSION__ !== 'undefined' ? __VERSION__ : 'latest';
+
+/** @type {LoaderObject} */
+export const ShapeLoader = {
+  id: 'shp',
+  name: 'Shapefile',
+  category: 'geometry',
+  version: VERSION,
+  extensions: ['shp'],
+  mimeTypes: [],
+  options: {
+    shapefile: {}
+  },
+  parse: parseShapefile
+};
+
+async function parseShapefile(arrayBuffer, options, context) {
+  const {parse, fetch, baseName} = context;
+
+  // Attempt a parallel load of the small sidecar files
+  // TODO: Can these extensions be both lower and uppercase? Match the main extension?
+  const shxResponse = fetch(`${baseName}.shx`);
+  const cpgResponse = fetch(`${baseName}.cpg`);
+  const prjResponse = fetch(`${baseName}.prj`);
+  await Promise.all([shxResponse, cpgResponse, prjResponse]);
```

Comment:
I hadn't seen this pattern of not assigning the result of `Promise.all` before... I love it

Reply:
Hmm yes we still need to await the responses after the Promise.all, I'll fix that.